processing/src/main/java/org/apache/druid/timeline/partition/OvershadowableManager.java

```diff
@@ -399,13 +395,20 @@ boolean isOvershadowedByVisibleGroup(RootPartitionRange partitionRange, short mi
    * A RootPartitionRange is smaller than a partitionId if {@link RootPartitionRange#startPartitionId} < partitionId.
    */
   private Iterator<Entry<RootPartitionRange, Short2ObjectSortedMap<AtomicUpdateGroup<T>>>> entryIteratorSmallerThan(
-      short partitionId,
+      int partitionId,
       TreeMap<RootPartitionRange, Short2ObjectSortedMap<AtomicUpdateGroup<T>>> stateMap
   )
   {
-    final RootPartitionRange lowFench = new RootPartitionRange((short) 0, (short) 0);
+    final RootPartitionRange lowFence = new RootPartitionRange(0, 0);
     final RootPartitionRange highFence = new RootPartitionRange(partitionId, partitionId);
-    return stateMap.subMap(lowFench, false, highFence, false).descendingMap().entrySet().iterator();
+    Iterator<Entry<RootPartitionRange, Short2ObjectSortedMap<AtomicUpdateGroup<T>>>> subItr;
+    try {
+      subItr = stateMap.subMap(lowFence, false, highFence, false).descendingMap().entrySet().iterator();
+    }
+    catch (IllegalArgumentException e) {
+      throw new IAE("Illegal partitionId %s", highFence);
```

Comment:
We should throw `DruidException.defensive` here.
You can read about the druid way of exceptions : https://github.com/apache/druid/pull/14004